infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -42,6 +43,18 @@ public abstract class SenderBufferManager {
      */
     public static final String config_file = "/config/corfu/corfu_replication_config.properties";
 
+    /*
+     * Back of napkin math to derive sleep time (per message):
+     *  - Assume payload size of 15MB
+     *  - Queue holds ~50 messages before OOM risk
+     *  -> MAX_SLEEP_TIME_MS = 1 min to process 50 messages = 60000 / 50 = 1200 + buffer = 1350ms
+     *  -> INITIAL_SLEEP_TIME_MS = 20% of max time
+     *  -> SLEEP_TIME_INCREMENT_MS = 10% of max time
+     */
+    public static final long INITIAL_SLEEP_TIME_MS = 270;
+    public static final long MAX_SLEEP_TIME_MS = 1350;
+    public static final long SLEEP_TIME_INCREMENT_MS = 135;
```

Comment:
In the ideal case you only want one re-transmission on a request timeout. Since this workflow is not latency sensitive, I suggest starting with a higher timeout (i.e., seconds) to reduce the probability of retrying more than once.